sympy/polys/domains/domain.py

```diff
@@ -1277,7 +1277,51 @@ def log(self, a, b):
         raise NotImplementedError
 
     def sqrt(self, a):
-        """Returns square root of ``a``. """
+        """Returns a (possibly inexact) square root of ``a``.
+
+        Explanation
+        ===========
+        There is no universal definition of "inexact square root" for all
+        domains. It is not recommended to implement this method for domains
+        other then :ref:`ZZ`.
```

Comment:
Rather than call this an inexact square root it would be better to say that it returns `floot(sqrt(a))` but computed exactly. Perhaps it would be better known as `isqrt` since it is equivalent to `math.isqrt`, `gmpy2.isqrt` etc.